go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -1005,10 +1005,13 @@ func (bnc *BaseNetworkController) DeleteResourceCommon(objType reflect.Type, obj
 			return fmt.Errorf("could not cast obj of type %T to *knet.NetworkPolicy", obj)
 		}
 		netinfo, err := bnc.networkManager.GetActiveNetworkForNamespace(knp.Namespace)
-		if err != nil {
+		// The InvalidPrimaryNetworkError error is thrown when UDN is not found because
+		// it has been already deleted, so just proceed with deleting NetworkPolicy in
+		// such a scenario as well.
+		if err != nil && !errors.Is(err, util.NewInvalidPrimaryNetworkError(knp.Namespace)) {
```

Comment:
Actually can you change this please to use a util function ( like we do for the other error type):

```
func IsInvalidPrimaryNetworkError(err error) bool {
	var invalidPrimaryNetworkError *InvalidPrimaryNetworkError
	return errors.As(err, &invalidPrimaryNetworkError)
}
```